docs/lib/DEBUGGING.md

```diff
@@ -152,3 +184,54 @@ If you are using ST-LINK, please refer to the [STMicroelectronics website](http:
 
 If instead you want to connect your debugger to the JTAG port, you must wire the needed pins to your connector. This [guide](https://www.segger.com/admin/uploads/evalBoardDocs/AN00015_ConnectingJLinkToSTM32F429Discovery.pdf) explains how to do that in detail. The guide is specific to the J-Link connectors, but you can apply it to other connectors.
 
+## Debugging with Eclipse
+
+* Make sure [GNU ARM Eclipse plugins](http://gnuarmeclipse.sourceforge.net/updates) is installed, if not then install:
+
+  `help -> Install New software -> Add http://gnuarmeclipse.sourceforge.net/updates`
+
+### Eclipse PyOCD debugger configuration
+
+* Create a new *GDB PyOCD Debugging* configuration (or *GDB SEGGER J-Link Debugging* for J-Link debugger):
+
+  `Run -> Debug configurations -> New`
+  
+  ![](../img/new_pyocd_cfg.png)
+  
+* **Main** tab configuration
+  * Give the path to the .elf of the application in *C/C++ Application*.
+  * Set *Disable auto build* if your eclipse project does not build the application.
```

Comment:
Set *Disable auto build* in case your Eclipse project is not configured to build the application.

Reply:
I disagree. I think "if" is the better option here.

Reply:
I also prefer "if"